pkg/mads/v1/service/components.go

```diff
@@ -46,11 +46,11 @@ func (r *restReconcilerCallbacks) OnFetchRequest(ctx context.Context, request ut
 		return errors.Errorf("expecting a v3 Node, got: %v", nodei)
 	}
 
-	// only reconcile if there is not a valid response present
-	if !r.reconciler.NeedsReconciliation(node) {
-		return nil
-	}
-
+	// TODO we need to reconcile on demand because if we us MeshMetric we will only reconcile in watchdog for single client
```

Comment:
can you create an issue for this? and link here / add to code

Reply:
added